scylla-gdb.py

```diff
@@ -4197,12 +4197,13 @@ def invoke(self, arg, from_tty):
         region_ptr_type = gdb.lookup_type('logalloc::region').pointer()
         for table in all_tables(db):
             gdb.write('table %s:\n' % schema_ptr(table['_schema']).table_name())
-            compaction_group = std_unique_ptr(table["_compaction_group"]).get()
-            memtable_list = seastar_lw_shared_ptr(compaction_group['_memtables']).get()
-            for mt_ptr in std_vector(memtable_list['_memtables']):
-                mt = seastar_lw_shared_ptr(mt_ptr).get()
-                reg = lsa_region(mt.cast(region_ptr_type))
-                gdb.write('  (memtable*) 0x%x: total=%d, used=%d, free=%d, flushed=%d\n' % (mt, reg.total(), reg.used(), reg.free(), mt['_flushed_memory']))
+            for cg_ptr in std_vector(table["_compaction_groups"]):
+                compaction_group = std_unique_ptr(cg_ptr).get()
+                memtable_list = seastar_lw_shared_ptr(compaction_group['_memtables']).get()
+                for mt_ptr in std_vector(memtable_list['_memtables']):
+                    mt = seastar_lw_shared_ptr(mt_ptr).get()
+                    reg = lsa_region(mt.cast(region_ptr_type))
+                    gdb.write('  (memtable*) 0x%x: total=%d, used=%d, free=%d, flushed=%d\n' % (mt, reg.total(), reg.used(), reg.free(), mt['_flushed_memory']))
```

Comment:
Please retain compatibility with older Scyllas. Although I see we already lost compatibility with pre-compaction-group scylla.

@denesb I don't think we can win this fight, since the tests don't care about older scylla.

Reply:
Yes, we usually try to enforce compatibility at review time and fix those that still sneak through later.

Reply:
will fix.